tests/Base.php

```diff
@@ -462,6 +462,85 @@ public function testDeprecatedMethods(): void
         self::assertEquals('{"hello":"world"}', $response['body']);
     }
 
+    public function testResponseChunkedSimple(): void
+    {
+        $body = '';
+        $chunks = 0;
+        $response = Client::execute(body: 'Hello', headers: ['x-action' => 'responseChunkedSimple'], callback: function($chunk) use(&$body, &$chunks) {
+            $body .= $chunk;
+            $chunks += 1;
+        });
+
+        self::assertEquals(200, $response['code']);
+        self::assertEquals('OK1OK2', $body);
+        self::assertGreaterThanOrEqual(2, $chunks);
+        self::assertEmpty(Client::getLogs($response['headers']['x-open-runtimes-log-id']));
+        self::assertEmpty(Client::getErrors($response['headers']['x-open-runtimes-log-id']));
+    }
+
+    public function testResponseChunkedComplex(): void
+    {
+        $body = '';
+        $chunks = 0;
+        $response = Client::execute(body: 'Hello', headers: ['x-action' => 'responseChunkedComplex'], callback: function($chunk) use(&$body, &$chunks) {
+            $body .= $chunk;
+            $chunks += 1;
+        });
+
+        self::assertEquals(201, $response['code']);
+        self::assertStringContainsString('Start', $body);
+        self::assertStringContainsString('Step1', $body);
+        self::assertStringContainsString('{"step2":true}', $body);
+        self::assertStringContainsString(\hex2bin('0123456789abcdef'), $body);
+        self::assertGreaterThanOrEqual(3, $chunks);
+        self::assertEmpty(Client::getLogs($response['headers']['x-open-runtimes-log-id']));
+        self::assertEmpty(Client::getErrors($response['headers']['x-open-runtimes-log-id']));
+        self::assertEquals('end', $response['headers']['x-trainer-header']);
+        self::assertEquals('start', $response['headers']['x-start-header']);
+    }
+
+    public function testResponseChunkedErrorStartDouble(): void
+    {
+        $body = '';
+        $chunks = 0;
+        $response = Client::execute(body: 'Hello', headers: ['x-action' => 'responseChunkedErrorStartDouble'], callback: function($chunk) use(&$body, &$chunks) {
+            $body .= $chunk;
+            $chunks += 1;
+        });
+
+        self::assertEquals(200, $response['code']);
```

Comment:
Wrong, it's 200 becasue request started chunk response. In which case status code is set during start, when it's set to 200 because  at start all is good.